source/linux/Foundational_Components/System_Security/Auth_boot.rst

```diff
@@ -0,0 +1,203 @@
+
+The root filesystem is hashed block by block, creating a hash tree (Merkle tree). The root hash of the hash tree is signed by a trusted key. During boot, the kernel verifies the hash tree before mounting the root filesystem. If any block is modified, the hash verification will fail, preventing tampered data from being used.
+
+While dm-verity guarantees data integrity, it does not promise confidentiality and works only on read-only filesystems.
+
+dm-crypt
+--------
+
+dm-crypt is a device-mapper target used for transparent disk encryption. It ensures data confidentiality by encrypting the entire partition or block device.
+
+A user provides an encryption key (stored securely in a TPM or entered manually). dm-crypt encrypts each block before writing it to disk. When reading data, dm-crypt decrypts blocks on the fly. Only authorized users with the correct key can access the decrypted data.
+
+Setup
+*****
+
+.. Image:: /images/Auth_default_bootflow.png
+     :align: center
+
+.. note::
+
+   A new Yocto layer is in the works to automate all of the below steps
+
+The following steps describe how to build user-space tools and configuration on Yocto. Please use :ref:`Processor SDK - Building the SDK with Yocto <building-the-sdk-with-yocto>` as reference.
+
+1. Use the latest :ref:`oe-config file <yocto-layer-configuration>`. Edit :file:`build/local.conf` to include support for *dm-crypt* and *dm-verity*:
+
+   .. code-block:: console
+
+      IMAGE_INSTALL:append = " cryptsetup lvm2"
+
+2. Build the Linux Kernel with support for these configs:
+
+   .. code-block:: kconfig
+
+      CONFIG_BLK_DEV_DM=y
+      CONFIG_DM_CRYPT=y
+      CONFIG_DM_VERITY=y
+
+   These configs can be added using a separate .cfg file or the kernel can be edited using
+
+   .. code-block::
+
+      MACHINE=<machine> bitbake -c menuconfig linux-ti-staging
+
+3. Build the default image:
+
+   .. code-block:: console
+
+      MACHINE=<machine> bitbake -k tisdk-default-image
+
+4. For this demo, the root filesystem is copied from the rootfs into the encrypted partition on 32GB+ SD card. Hence, the SD card needs to be partitioned accordingly. It is recommended to create 2 additional ext4 partitions bringing the total to 4 partitions:
+
+   +-----------------+----------------+-------+--------------+
+   | Partition Label | /dev partition | Size  |   Comments   |
+   +=================+================+=======+==============+
+   |      boot       | /dev/mmcblk1p1 | 128MB |   Default    |
+   |      root       | /dev/mmcblk1p2 |  10GB |   Default    |
+   |      crypt      | /dev/mmcblk1p3 |  10GB | Same as root |
+   |     verity      | /dev/mmcblk1p4 |  1GB  | 10% of crypt |
+   +-----------------+----------------+-------+--------------+
+
+4. Boot into the target, and run the following targets to setup the crypt and verity partitions
+
+   .. code-block:: console
+
+      # Create a random pass key
+      dd if=/dev/urandom of=/home/pass_key bs=64 count=1
```

Comment:
May want to use a similar keyfile method Arch recommends, for portability:
https://wiki.archlinux.org/title/Dm-crypt/Device_encryption#Random_characters